### ISSUE TITLE
Upload sample field scanner data to Clowder

### ISSUE THREAD

**Issue by dlebauer** [terraref/computing-pipeline#66]
We have two sets of sample data from the field scanners: https://uofi.box.com/s/c9ngkwi3xvtfr0ei5zfzgf4jya0dmrlc

Would be good to start testing out the use of Clowder for distribution of these datasets. 

* [x] upload the .zip files
* [x] unzip and upload the directories

    perhaps there should be one clowder dataset per scanner x timepoint and collection per date (?)
* [x] provide API calls to query and download these datasets

**Comment by max-zilla** [comment]
I have an initial batch upload in the Terra dev instance:
http://141.142.209.122/clowder/collections
I will need to modify the script to upload to the production instance (141.142.208.144) because it doesn't currently have the code to push file+JSON metadata at once. But this will let you see the structure - please let me know if you can see it.

A couple notes:

- We have a model one step up from collections - Spaces. It could make sense to have each date be a Space, each sensor be a Collection in that space, and each timestamp be a dataset within that collection. Perhaps we'll need Spaces for higher levels of organization.

- I did not upload the .json file - I just attached the contents as metadata to the .bin file. When looking at a file's metadata, one can use this button to download the metadata in JSONLD format, so I don't think we lose anything doing so: 
![image](https://cloud.githubusercontent.com/assets/14166971/13112686/7facaa7a-d551-11e5-9b7a-8d74871222e9.png)

- I kept the Box name for the MovingSensor dataset as "Feb 09" even though the actual data is from 02/10 - just for coherency between Box and Clowder.

**Comment by robkooper** [comment]
I would like to work and see if we can use the new folder in dataset functionality (https://opensource.ncsa.illinois.edu/jira/browse/CATS-367) that exists in clowder. We can add a function that will allow us to upload a zip file that would automatically be extracted to a bunch of folders/files inside a dataset.

**Comment by robkooper** [comment]
@max-zilla can you add the collection to a test space that all developers on the server belong to?

**Comment by max-zilla** [comment]
@robkooper created one and added you, Dora, Yan Liu, Noah. will add David L. and others once accounts are created.

**Comment by max-zilla** [comment]
In terms of programatically downloading files, there are a couple API calls one might want:

```
url = "http://http://141.142.209.122/clowder/"

GET /api/collections/list (get all collections in a list the user can access)
GET /api/collections/:coll_id/getDatasets (list all datasets in a given collection)
GET /api/datasets/:id/download (download dataset with given ID)
```

**Comment by dlebauer** [comment]
@max-zilla could you please convert that to a curl command that I can copy-paste to the command line?

**Comment by dlebauer** [comment]
Looking at the datasets at here are a few thoughts:

* often there are many files and one metadata file. We can assume that each high-level folder ( a lemnatec 'snapshot') will have one or more 'raw' data files and one .json metadata file (the Feb 09 had some xml, but these are deprecated)
* This set of files is the unit of interest - the files are not as useful alone (e.g. two images for stereo 3D reconstruction) .
  * how can we store it? Currently each individual 'file' is its own record 
  * will this be solved by folders within datasets?
* for derived files we could either leave them with the source files as we do in PlantCV or create a new 'dataset' for the derived images and transfer the relevant metadata. In some cases the meta-data will be used in the conversion step or embedded in the file, so it might not always be useful to just blindly copy it

**Comment by max-zilla** [comment]
You'll need to add your credentials but an example call and response:

```
curl -X GET -u USER:PASS http://141.142.209.122/clowder/api/collections/list 
```

returns:

```json
[
{"id":"56c49d0ae4b0627842cb03f1","collectionname":"2016 Feb 12 - V2","description":"Sample upload from https://uofi.app.box.com/s/c9ngkwi3xvtfr0ei5zfzgf4jya0dmrlc/1/6560260529","created":"Wed Feb 17 16:17:14 UTC 2016","thumbnail":"None","authorId":"mburnet2@illinois.edu"},
{"id":"56c49c44e4b0627842cb0336","collectionname":"2016 Feb 09 - V2","description":"Sample upload from https://uofi.app.box.com/s/c9ngkwi3xvtfr0ei5zfzgf4jya0dmrlc/1/6499761225","created":"Wed Feb 17 16:13:56 UTC 2016","thumbnail":"None","authorId":"mburnet2@illinois.edu"},
{"id":"56c48745e4b0627842cb007f","collectionname":"2016 Feb 09","description":"Sample upload from https://uofi.app.box.com/s/c9ngkwi3xvtfr0ei5zfzgf4jya0dmrlc/1/6499761225","created":"Wed Feb 17 14:44:21 UTC 2016","thumbnail":"None","authorId":"mburnet2@illinois.edu"},{"id":"56c48460e4b0627842cafe45","collectionname":"2016 Feb 12","description":"Sample upload from https://uofi.app.box.com/s/c9ngkwi3xvtfr0ei5zfzgf4jya0dmrlc/1/6560260529","created":"Wed Feb 17 14:32:00 UTC 2016","thumbnail":"None","authorId":"mburnet2@illinois.edu"}
]
```

From there, one could list datasets in one of these collections:

```
curl -X GET -u USER:PASS http://141.142.209.122/clowder/api/collections/56c49d0ae4b0627842cb03f1/getDatasets
```

returns:

```json
[
{"id":"56c49d12e4b0627842cb048b","name":"stereoTop","description":"","created":"Wed Feb 17 16:17:22 UTC 2016","thumbnail":"None","authorId":"mburnet2@illinois.edu"},
{"id":"56c49d0fe4b0627842cb0451","name":"ps2Top","description":"","created":"Wed Feb 17 16:17:19 UTC 2016","thumbnail":"None","authorId":"mburnet2@illinois.edu"},
{"id":"56c49d0ae4b0627842cb03f3","name":"flirTop","description":"","created":"Wed Feb 17 16:17:14 UTC 2016","thumbnail":"None","authorId":"mburnet2@illinois.edu"}]
```

From there one can download a dataset:
```
curl -X GET -u USER:PASS http://141.142.209.122/clowder/api/datasets/56c49d12e4b0627842cb048b/download > file.zip
```

**Comment by dlebauer** [comment]
got it. where `USER` = email address. Can I use an API key instead of a password?

**Comment by max-zilla** [comment]
I've created a V3 space and invited people to it:
http://141.142.209.122/clowder/spaces/56c632d3e4b0627842cb26fd

This is slightly smarter about the foldername-derived metadata in naming datasets. I also modified my script to create collections for sensor height and named group (parsed from folder name) - it's mostly illustrative.

Also fixed bug that only expected one file and one .json per folder. Now there are some datasets out there with up to 102 .bin files in them.
